core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -36,26 +36,26 @@ sealed trait Chunk[+A] extends ChunkLike[A] { self =>
   /**
    * Appends an element to the chunk
    */
-  final def +[A1 >: A](a: A1): NonEmptyChunk[A1] =
-    self match {
-      case Chunk.Empty          => Chunk.single(a)
-      case ne: NonEmptyChunk[A] => Chunk.concat(ne, Chunk.single(a))
-    }
+  final def +[A1 >: A](a: A1): Chunk[A1] =
```

Comment:
this could still return NonEmptyChunk

Reply:
We could but there is potentially a little extra overhead in creating the `NonEmptyChunk` which I thought you might not want if you were doing repeated appends and were doing high performance stuff where you didn't care about whether it was empty or not (e.g. you're just calling `foreach` on it). So right now you kind of stay in the `Chunk` world unless you explicitly go to the `NonEmptyChunk` world, and once you're there we preserve that information whenever possible. But we could definitely change as you suggest.